src/OrchardCore.Modules/OrchardCore.ContentFields/Services/DefaultContentPickerResultProvider.cs

```diff
@@ -27,7 +27,7 @@ public async Task<IEnumerable<ContentPickerResult>> Search(ContentPickerSearchCo
 
             if (!string.IsNullOrEmpty(searchContext.Query))
             {
-                query.With<ContentItemIndex>(x => x.DisplayText.Contains(searchContext.Query));
+                query.With<ContentItemIndex>(x => x.ToString().Contains(searchContext.Query));
```

Comment:
This won't work, you need it the way you had it before (testing for both display text and content type)